src/config.rs

```diff
@@ -144,15 +168,16 @@ pub fn save_config(config: &Config) -> Result<()> {
 /// use dead_man_switch::config::load_or_initialize_config;
 /// let config = load_or_initialize_config().unwrap();
 /// ```
-pub fn load_or_initialize_config() -> Result<Config> {
+pub fn load_or_initialize_config() -> Result<Config, ConfigError> {
     let config_path = config_path()?;
     if !config_path.exists() {
         let config = Config::default();
         save_config(&config)?;
         Ok(config)
     } else {
-        let config = fs::read_to_string(&config_path)?;
-        let config: Config = toml::from_str(&config)?;
+        // Tries to read the config file
+        let config = fs::read_to_string(&config_path).map_err(|_| ConfigError::ReadConfigFile)?;
+        let config: Config = toml::from_str(&config).map_err(|_| ConfigError::ParseConfigFile)?;
```

Comment:
Same as above, we don't need to reinvent the whefl. All of these returns Errors that we can convert from.